### PR TITLE
Add --keep-intermediates CLI option

### LINKED ISSUE #105: Add --keep-intermediates option to ``infer`` command
Many times we'll just want to run the full chain with default-ish params, but also want to keep the ancestors files lying around.

### PR DESCRIPTION
**Pull request by hyanwong**
Thought I would fix this niggle while fiddling with the CLI files. Fixes #105

### CODE REVIEW

File: tests/test_cli.py [hunk 1]
```diff
@@ -226,6 +226,37 @@ def test_augment_ancestors(self):
...
+                "infer",
+                self.sample_file,
+                "--keep-intermediates",
+                "-a",
+                output_anc,
+                "-A",
+                output_anc_ts,
+            ]
+        )
+        assert os.path.exists(output_anc)
```
Comment (benjeffery) [inline tests/test_cli.py:256]: Can we go one better here and check that the right files go to the right place? Maybe by loading the ts?

**Review by benjeffery — APPROVED** [review]
Sorry for the delayed review here @hyanwong! Just one comment on testing.

File: tests/test_cli.py (hunk 1 above)
Comment (hyanwong) [reply tests/test_cli.py:256]: Good call. I loaded both files and checked num_ancestors/ samples

**Comment by hyanwong** [comment]
I think we can merge this, right @benjeffery ?

**Comment by benjeffery** [comment]
> I think we can merge this, right @benjeffery ?

I'm happy for it - was hoping for a nod from JK - but I think his plate is full. Let's merge and ask for forgiveness if necessary ;)

**Comment by benjeffery** [comment]
Gah #788. Will fix.

**Comment by hyanwong** [comment]
> Will fix.

Thank you for all the hard work you are doing!

**Comment by benjeffery** [comment]
@mergifyio rebase

**Pull request merged**